src/Flysystem/LocalFilesystemAdapter.php

```diff
@@ -27,4 +47,75 @@ public function prefixPath(string $path): string
     {
         return $this->pathPrefixer->prefixPath($path);
     }
+
+    /**
+     * List the contents of a path by glob.
+     *
+     * @return iterable<StorageAttributes>
+     */
+    public function listContentsByGlob(string $folder, string $fileGlob, bool $deep = FilesystemReader::LIST_SHALLOW): iterable
+    {
+        $basePath = $this->prefixPath($folder);
+        if (!is_dir($basePath)) {
+            return [];
+        }
+
+        // Use the glob to match the base path first.
+        yield from $this->findByGlob(Path::join($basePath, $fileGlob));
+
+        // If we're not checking subdirectories, just return out.
+        if (!$deep) {
+            return;
+        }
+
+        // Check contents of subdirectories recursively.
+        $iterator = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(
+                $basePath,
+                FilesystemIterator::SKIP_DOTS | FilesystemIterator::KEY_AS_PATHNAME | FilesystemIterator::CURRENT_AS_FILEINFO
+            ),
+            RecursiveIteratorIterator::SELF_FIRST
+        );
+        /** @var SplFileInfo $info */
+        foreach ($iterator as $path => $info) {
+            if ($info->isDir()) {
+                yield from $this->findByGlob(Path::join($path, $fileGlob));
+            }
+        }
+    }
+
+    private function findByGlob(string $glob): Generator
+    {
+        $iterator = new GlobIterator($glob, FilesystemIterator::SKIP_DOTS);
+        foreach ($iterator as $fileInfo) {
+            // NOTE: The contents of this loop are directly copied from listContents()
```

Comment:
Then should have a shared private method to share the code?

Reply:
We don't control the code in `listContents()`. That's Flysystem.